src/aics-image-viewer/components/App/index.tsx

```diff
@@ -969,26 +949,24 @@ export default class App extends React.Component<AppProps, AppState> {
     };
   }
 
-  // TODO rendering algorithm should become an enum
-  onChangeRenderingAlgorithm(newAlgorithm: string) {
+  onChangeRenderingAlgorithm(newAlgorithm: RenderMode) {
     const { userSelections } = this.state;
     // already set
     if (userSelections[newAlgorithm]) {
       return;
     }
     this.setUserSelectionsInState({
-      pathTrace: newAlgorithm === PATH_TRACE,
-      maxProject: newAlgorithm === MAX_PROJECT,
-      autorotate: newAlgorithm === PATH_TRACE ? false : userSelections.autorotate,
+      pathTrace: newAlgorithm === RenderMode.pathTrace,
+      maxProject: newAlgorithm === RenderMode.maxProject,
+      autorotate: newAlgorithm === RenderMode.pathTrace ? false : userSelections.autorotate,
     });
-    this.handleChangeToImage(PATH_TRACE, newAlgorithm === PATH_TRACE);
-    this.handleChangeToImage(MAX_PROJECT, newAlgorithm === MAX_PROJECT);
+    this.handleChangeUserSelection("pathTrace", newAlgorithm === RenderMode.pathTrace);
+    this.handleChangeUserSelection("maxProject", newAlgorithm === RenderMode.maxProject);
```

Comment:
I'm slightly dubious that we gained all that much when I see explicit quoted strings being passed in, but then I remember that it's type checked!  A casual reader of this code would not know that at a glance.

Reply:
You're right to be at least a little bit dubious, since proper type-checking for these generic functions is basically dependent on callers passing actual string literals rather than variables. It's possible e.g. for someone to invoke
```
handleChangeUserSelection<ChannelStateKey>(key, value);
```
and have the type of `value` be assessed as the union of all types that appear in `ChannelState`, regardless of which `key` is actually being changed.

Anyways, should the type-checking behavior of this and similar functions be documented somewhere?

Reply:
I still wonder if it would be better to just let each change function have its own well typed arguments. It's going to move closer and closer to just having explicit functions.
```
handleChangeUserSelectionMaxProject(isMaxProject); // typed as boolean because that's what maxproject wants
```
or
```
handleChangeUserSelection["maxProject"](isMaxProject); 
```
where handleChangeUserSelection is the object that holds on to all the functions, and is allowed to have many functions all with different signatures

Reply:
The main convenience I think we're getting from having a single way into these maps of handlers is having not-undefined guards for `view3d` and `image` in a single place rather than duplicated across a bunch of handlers. But there are good potential solutions for that too. e.g. in `TFEditor` a number of important properties aren't set until `componentDidMount`, and eventually rather than guard these types everywhere they're needed I started defining them in the class interface with a `!:` for "this property will spend part of its lifecycle undefined, but will never be accessed before being given its initial value". If that's the case for `view3d` and `image` too then we can make that change in `AppState` and not have to worry about putting not-undefined checks all over the place. In that case I think it might make more sense to either pass the big handler map around or else just define individual handlers.